### ISSUE TITLE
vector_graphics 1.1.10 breaks widgets inside an opacity

### ISSUE THREAD

**Issue by vanlooverenkoen** [dnfield/vector_graphics#231]
@dnfield & @jason-simmons t[his change](https://github.com/dnfield/vector_graphics/pull/229) breaks our widgets. 

We have an opacity around our widget

```dart
   Opacity(
      opacity: _loading ? 0 : 1,
      child: Row(
        mainAxisSize: MainAxisSize.min,
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          if (widget.icon != null && widget.iconPosition == ButtonIconPosition.start) ...[
            _buildIcon(layer),
            SizedBox(width: 8),
          ],
          Text(
            'Some text',
          ),
          if (widget.icon != null && widget.iconPosition == ButtonIconPosition.end) ...[
            SizedBox(width: 8),
            _buildIcon(layer),
          ],
        ],
      ),
    );
```

This results in something like this now: 

![image](https://github.com/dnfield/vector_graphics/assets/21172855/c93205b5-5ed0-4be7-b37f-40622085f303)

Before this change our widget looked like this:

![image](https://github.com/dnfield/vector_graphics/assets/21172855/46c8d9c4-267d-4e21-9ed8-529e1c2f47b0)

(don't look at the size, the text is the important part that is missing)

Maybe I'm missing something, but I don't think we are doing something wrong here.

**Comment by vanlooverenkoen** [comment]
This can be a workaround and maybe a better solution for my case. But I feel like it should still be possible to place an opacity around an svg/vector_graphics widget

```dart
   Visibility.maintain([
      visible: !_loading,
      child: Row(
        mainAxisSize: MainAxisSize.min,
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          if (widget.icon != null && widget.iconPosition == ButtonIconPosition.start) ...[
            _buildIcon(layer),
            SizedBox(width: 8),
          ],
          Text(
            'Some text',
          ),
          if (widget.icon != null && widget.iconPosition == ButtonIconPosition.end) ...[
            SizedBox(width: 8),
            _buildIcon(layer),
          ],
        ],
      ),
    );
```

**Comment by furaiev** [comment]
Hello, we have failed golden tests with 1.1.10. Visually no changes but the reported diff is 0.04%

**Comment by dnfield** [comment]
Argh. I believe we're missing a save/restore.

**Comment by dnfield** [comment]
> Hello, we have failed golden tests with 1.1.10. Visually no changes but the reported diff is 0.04%

This small of a diff is probably not a bug.

**Comment by dnfield** [comment]
@vanlooverenkoen can you share a more complete sample, ideally pared down a bit to just the smallest needed to reproduce the issue? I'm pretty sure we need another save but want to make sure it fixes the case you're dealing with.

Alternatively, can you add a `context.canvas.save();` before the `clipRect` call and see if that fixes things?

**Comment by dnfield** [comment]
FWIW, Visibility is a better widget to use for your usecase than Opacity :)

**Comment by dnfield** [comment]
I retracted 1.1.10 and uploaded the fix in 1.1.10+1. Thanks for the report.
